docs/daisy-migrating-existing-workflows.md

```diff
@@ -0,0 +1,110 @@
+
+You can use this partial URL in the "SourceImage" field in a "CreateDisks" Daisy
+step, or anywhere else that Daisy accepts partial URLs. By using an image
+family, you ensure that you will always run your workflow on the most up-to-date
+base image available.
+
+## Translating your Current Workflow
+
+We recommend that you write down every step you perform to prepare your image,
+in the order they are performed. This will be different for every image, but it
+usually involves some combination of downloading and running installers,
+building code, invoking package managers, and writing to configuration files.
+
+The biggest thing to keep in mind is that Daisy only allows you to run one
+script on startup. This script can download and run other scripts, but it must
+be the one to do it. Once you have listed all of the steps you take to set up
+your image, write a script that performs all of them.
+
+All of the files that your startup script needs to access (which are not already
+present on the machine) should be listed in the "Sources" field of your Daisy
+workflow. Sources can be paths to local files, or to files in a Storage bucket
+using the URL format `gs://<bucket>/<file>`.
+
+Files listed in the "Sources" field of your Daisy workflow are all copied to a
+special location in Google Cloud Storage. For more information on getting the
+URL of this location and downloading files from it, see [Passing Data to
+Instances](daisy-passing-data.md).
+
+All Windows image creation startup scripts should end with a call to
+`gcesysprep.bat`. All other OS scripts should end by shutting down the instance.
+
+## A Typical Workflow
+
+The typical Daisy image creation workflow has five steps:
+
+    1. A CreateDisks step which creates at least one disk. The SourceImage for
+       one of these disks should be your base image.
+    2. A CreateInstances step which creates an instance out of the disks the
+       previous step created. The first disk in this step's `Disks` field must
+       be bootable. This step also specifies the StartupScript.
+    3. A WaitForInstancesSignal step which waits for the startup script to
+       complete. WaitForInstancesSignal has a field called `Stopped` to listen
+       for the instance shutting down. It can also be made to listen for
+       specific output on the serial port.
+    4. A CreateImages step which actually creates an image from the disk once
+       the instance has shut down. Make sure you specify `NoCleanup` on this
+       step, or your image will be deleted when the workflow ends!
+    5. A DeleteResources step that deletes the instance you created in the
+       CreateInstances step.
+
+Each step should depend on the step before it. Use the `Dependencies` field of
+your Daisy workflow to specify dependencies. If no dependencies are specified,
+then Daisy will try to run all the steps in parallel, which may give you some
+*interesting* results.
```

Comment:
We have validation that will catch things like trying to start an instance from a disk that that step does not depend on

Reply:
Added a note about that behavior.